packages/jellyfish-api-core/__tests__/category/account/getPendingFutureSwaps.test.ts

```diff
@@ -0,0 +1,333 @@
+
+      // Generate 1 more block to settle
+      await testing.generate(1)
+
+      // Call getpendingfutureswaps after futureswap is settled
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: []
+        })
+      }
+    })
+
+    it('Should getPendingFutureSwaps if futureswap DUSD for TSLA', async () => {
+      // Call getpendingfutureswaps before performing futureswap
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: []
+        })
+      }
+
+      // futureswap
+      await testing.container.call('futureswap', [tslaAddress, '1@DUSD', 'TSLA'])
+      await testing.generate(1)
+
+      // Call getpendingfutureswaps after performing futureswap
+      const pendingFutureSwaps1 = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+      expect(pendingFutureSwaps1).toStrictEqual({
+        owner: tslaAddress,
+        values: [
+          {
+            source: '1.00000000@DUSD',
+            destination: 'TSLA'
+          }
+        ]
+      })
+
+      // Wait for 1 block before the next settle block
+      const nextSettleBlock = await getNextSettleBlock()
+      await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount() - 1)
+
+      const pendingFutureSwaps2 = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+      expect(pendingFutureSwaps2).toStrictEqual(pendingFutureSwaps1) // Nothing change
+
+      // Generate 1 more block to settle
+      await testing.generate(1)
+
+      // Call getpendingfutureswaps after futureswap is settled
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: []
+        })
+      }
+    })
+  })
+
+  describe('Multiple futureswaps', () => {
+    it('Should getPendingFutureSwaps if futureswap TSLA for DUSD', async () => {
+      // Call getpendingfutureswaps before performing futureswap
+      {
+        const pendingFutureSwaps = await testing.rpc.account.getPendingFutureSwaps(tslaAddress)
+        expect(pendingFutureSwaps).toStrictEqual({
+          owner: tslaAddress,
+          values: []
+        })
+      }
+
+      // futureswap 4 times
+      await testing.container.call('futureswap', [tslaAddress, '0.7@TSLA'])
+      await testing.generate(1)
+
+      await testing.container.call('futureswap', [tslaAddress, '0.6@TSLA'])
+      await testing.generate(1)
+
+      await testing.container.call('futureswap', [tslaAddress, '0.5@TSLA'])
+      await testing.generate(1)
+
+      await testing.container.call('futureswap', [tslaAddress, '0.2@TSLA'])
+      await testing.generate(1)
```

Comment:
one more test request - check the `getPendingFutureSwaps` result on same block testing
```
futureswap(1)
futureswap(2)
gen(1)
```